### ISSUE TITLE
[📖] CSS Modules and useStyles$() hook approach incompatibility

### ISSUE THREAD

**Issue by shukklin** [QwikDev/qwik#7145]
### Suggestion

Original docs URL are:
https://qwik.dev/docs/components/styles/#css-modules
https://qwik.dev/docs/components/styles/#usestyles

The context:
1. We have several Qwik repositories - UI Kit and SSR application.
2. We build the UI Kit using Vite with separate index (.js, .css) files for each components to support further tree-shaking in the SSR application.
3. Then we use these builded files in the SSR application and link styles using the useStyles$() hook to support tree-shaking. Because if we import only pure CSS modules without processing it with useStyles$() hook, we will get one large CSS file bundled, which is not good for a large amount of CSS code (currently we have about 840KB CSS bundle size code in total).

We have found that Qwik does not support the use of CSS Modules together with useStyles$()/useStylesScoped$() hooks to load only required CSS rules on the page, as the styles loaded from CSS Modules don't tree-shake (look at the screenshot below with identical styles in different chunks). This leads to CSS code duplication and increased bundle size.

<img width="1006" alt="image" src="https://github.com/user-attachments/assets/d37fd55e-e49a-42fd-af25-dee807686855">

As it turned out, useStyles$()/useStylesScoped$ hooks only supports plain CSS, but not CSS Modules.

I think this point should be better reflected in the documentation, because at the moment there is nothing about it and it is not entirely clear.

**Comment by shairez** [comment]
@Varixo is it the same in V2?

**Comment by Varixo** [comment]
> [@Varixo](https://github.com/Varixo) is it the same in V2?

yes, probably yes

**Comment by wmertens** [comment]
@shukklin so you just `import myCss from 'my.modules.css?inline'`? Can you provide a repro?

**Comment by shukklin** [comment]
@wmertens yep, sure. Here is the [link to the repro](https://stackblitz.com/edit/github-b9xsukss-uxwhjz7k?file=src%2Froutes%2Findex.tsx) (go to the `./src/routes/index.tsx` file).

As I said, we have a separate UI Kit library package where we use CSS modules. After building package, we get a component.tsx file and a index.css file with CSS modules classnames (in the repro it is called `<ModuleButton />`). 

Then in the Qwik application we enable the `<ModuleButton />` component styles (`import styles from ../ModuleButton/index.css?inline`) via the `useStyles$(styles)` hook to enable tree-shaking instead of applying them globally (`import './../ModuleButton/index.css'`). But as you can see in the repro, if we have several components on a single page that using the same component from UI Kit with Module CSS, then the styles are duplicated in HTML, as shown in the screenshot below.

<img width="1143" height="882" alt="Image" src="https://github.com/user-attachments/assets/049480c0-bd62-4225-8373-d37f3eca8d35" />

**Comment by wmertens** [comment]
@shukklin as you can see, each style has its own hash. The reason is that the imported css is converted to a QRL in every place that useStyles$ is used, and we don't do QRL deduplication.

The fix is to create the QRL for the styles in one place, and use that with useStylesQrl.

e.g.
```tsx
import styles from "../ModuleButton/index.css?inline"

export const stylesQrl = $(styles)

...
useStylesQrl(stylesQrl)
```

**Comment by wmertens** [comment]
actually, fixed in #8436 😁
